packages/sanity/src/core/util/draftUtils.ts

```diff
@@ -74,26 +83,40 @@ export function getIdPair(
     draftId: getDraftId(id),
     ...(version
       ? {
-          versionId: id.startsWith(`${version}.`) ? id : [version, getPublishedId(id)].join('.'),
+          versionId: isVersionId(id) ? id : getVersionId(id, version),
         }
       : {}),
   }
 }
 
 /** @internal */
 export function isPublishedId(id: string): id is PublishedId {
-  return !isDraftId(id)
+  return !isDraftId(id) && !isVersionId(id)
 }
 
 /** @internal */
 export function getDraftId(id: string): DraftId {
   return isDraftId(id) ? id : ((DRAFTS_PREFIX + id) as DraftId)
 }
 
+/**  @internal */
+export function getVersionId(id: string, bundle: string): string {
+  if (isVersionId(id)) {
+    const [version, bundleName, ...publishedId] = id.split('.')
+    if (bundleName === bundle) return id
+    return `${VERSION_PREFIX}${bundle}${PATH_SEPARATOR}${publishedId}`
+  }
+
+  const publishedId = getPublishedId(id)
+
+  return `${VERSION_PREFIX}${bundle}${PATH_SEPARATOR}${publishedId}`
+}
+
 /** @internal */
-export function getPublishedId(id: string, isVersion?: boolean): PublishedId {
-  if (isVersion) {
-    return id.split(PATH_SEPARATOR).slice(1).join(PATH_SEPARATOR) as PublishedId
+export function getPublishedId(id: string): PublishedId {
+  if (isVersionId(id)) {
+    // always return the last segment of the id
+    return id.split('.').pop() as PublishedId
```

Comment:
are we assuming we will never have ids on the format `versions.foo.bar.baz`? Would it not be better if this worked the same way as for draft ids below, i.e. returning everything that comes after `versions.<bundleId>.`?

Reply:
According to the working contract with Content Lake, and already assumed in the temporary query we now use instead of `sanity::versionOf`, I'd agree that `PublishedId` is everything defined after the `versions.<bundleSlug>.`

Reply:
ah my bad :)
I always assumed that the published Id was always just the single string that is at the end (with the rest being bundle info) especially since that is how the versionOf was first introduced initially (in a doc like `versions.bundle-summer.agot-id`), you need to send `agot-id` like versionOf(agot-id) to get all the versions associated with that id. If it had sent `bundle-summer.ago-id` I'm not sure that would have worked.

But that might have changed.
Keep in mind the temporary query was just to get it working so I could debug the code, not as a reflection of decisions made around the ids.

Reply:
Spoke on slack about the change, will be making it :)

Reply:
here we go https://github.com/sanity-io/sanity/pull/7357/commits/f4350541c0ac42a651dbdeb4017cbc50826f2bdf